backend/src/main/java/com/ody/route/service/ApiCallService.java

```diff
@@ -20,42 +21,48 @@ public class ApiCallService {
 
     private final ApiCallRepository apiCallRepository;
 
-    public ApiCallCountResponse countOdsayApiCall() {
-        Optional<ApiCall> apiCall = apiCallRepository.findFirstByDateAndClientType(
-                LocalDate.now(),
-                ClientType.ODSAY
-        );
-        return apiCall.map(call -> new ApiCallCountResponse(call.getCount()))
-                .orElseGet(() -> new ApiCallCountResponse(0));
-    }
+    public ApiCallCountResponse countApiCall(ClientType clientType) {
+        LocalDate end = LocalDate.now();
+        LocalDate start = clientType.determineResetDate(end);
+        List<ApiCall> apiCalls = apiCallRepository.findAllByDateBetweenAndClientType(start, end, clientType);
 
-    public ApiCallCountResponse countGoogleApiCall() {
-        LocalDate now = LocalDate.now();
-        List<ApiCall> apiCalls = apiCallRepository.findAllByClientTypeAndDateBetween(
-                ClientType.GOOGLE,
-                now.withDayOfMonth(1),
-                now
-        );
         int totalCount = apiCalls.stream()
                 .mapToInt(ApiCall::getCount)
                 .sum();
         return new ApiCallCountResponse(totalCount);
     }
 
     @Transactional
-    public ApiCall increaseCountByRouteClient(RouteClient routeClient) {
-        ClientType clientType = routeClient.getClientType();
-        return apiCallRepository.findFirstByDateAndClientType(LocalDate.now(), clientType)
-                .map(this::updateCount)
-                .orElseGet(() -> saveInitialCount(clientType));
+    public void increaseCountByClientType(ClientType clientType) {
+        ApiCall apiCall = findOrSaveFirstByClientTypeAndDate(clientType);
+        apiCall.increaseCount();
     }
 
-    private ApiCall updateCount(ApiCall apiCall) {
-        apiCall.increaseCount();
-        return apiCall;
+    private ApiCall findOrSaveFirstByClientTypeAndDate(ClientType clientType) {
+        Optional<ApiCall> apiCall = apiCallRepository.findFirstByDateAndClientType(LocalDate.now(), clientType);
```

Comment:
이 부분 Optional 체이닝 활용해도 좋을 것 같아요!

```suggestion
        return  apiCallRepository.findFirstByDateAndClientType(LocalDate.now(), clientType)
        .orElseGet(~~)
```

Reply:
[반영] [df1bdcc](https://github.com/woowacourse-teams/2024-ody/pull/732/commits/df1bdcc4f99adb80d900acaf619cac63da2d7966)